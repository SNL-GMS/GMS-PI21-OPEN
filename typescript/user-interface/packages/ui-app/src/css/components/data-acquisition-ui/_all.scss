@import 'soh-vars';
@import 'soh-cell';
@import 'ag-theme-overrides.scss';
@import 'data-acquisition';
@import 'environment-history';
@import 'historical-trends-history';
@import 'soh-chart';
@import 'station-statistics';
@import 'soh-lag-missing';
@import 'soh-environment';
@import 'soh-overview';
@import 'soh-quiet-indicator';
@import 'capability-counts';
@import 'soh-map';
