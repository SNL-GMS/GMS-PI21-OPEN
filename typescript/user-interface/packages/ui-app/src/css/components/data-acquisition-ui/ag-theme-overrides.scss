.ag-theme-dark--soh {
  border-radius: $cell-cap-radius;

  .ag-header {
    background: transparent;
  }

  .ag-header-cell {
    .ag-react-container {
      width: 100%;
    }
  }
  .ag-header-viewport {
    background-color: var(--gms-prominent-background, $gms-prominent-background);
  }

  .ag-row {
    overflow: hidden;
    border-radius: $cell-cap-radius;
  }

  .ag-ltr .ag-cell:not(.ag-cell-focus).ag-cell-last-left-pinned,
  .ag-ltr .ag-row.ag-cell-last-left-pinned,
  .ag-horizontal-left-spacer {
    border-right: none;
  }

  &:not(.table--cell-selection-only) {
    .ag-row-hover .ag-cell {
      @include brighten;
    }
  }
  &.table--cell-selection-only {
    .ag-cell:hover {
      @include brighten;
    }
  }

  // TODO: Make this dynamic so it doesn't expect a fixed
  // number of columns
  .ag-row-selected {
    .ag-cell .table-cell {
      box-sizing: border-box;
      @include selectedBorder('top', 'bottom');
    }
    .ag-cell[aria-colindex='1'] .table-cell {
      @include selectedBorder('left');
    }
    .ag-cell .table-cell[data-col-position='last'] {
      @include selectedBorder('right');
    }
  }

  .ag-body-viewport,
  .ag-root-wrapper {
    background: transparent;
  }
}
