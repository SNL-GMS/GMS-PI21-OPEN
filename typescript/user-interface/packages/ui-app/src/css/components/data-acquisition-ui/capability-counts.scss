.stacked-counts {
  @include flexColumn;
  @include centerChildren;
  &__entry {
    padding: 0;
    margin: 0;
    line-height: 1em;
    padding-right: 0.25rem;
    margin-left: -0.25rem;
    font-weight: 400;
    opacity: 0.8;
    font-size: 0.875em;
    & + & {
      margin-top: 0.125rem;
    }
  }
}

// Color the marginal capability count yellow if it is inside of a red cell
[data-cell-status='bad'] .stacked-counts__entry[data-capability-status='marginal'] {
  color: $marginal-color;
}
