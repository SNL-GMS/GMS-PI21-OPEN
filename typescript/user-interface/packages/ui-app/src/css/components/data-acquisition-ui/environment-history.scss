.environment-history-display {
  &__chart-container {
    height: auto;
    width: 100%;
    overflow-x: auto;
  }
  .external-axis {
    min-height: 50px;
    background-color: var(--gms-prominent-background, $gms-prominent-background);
    border-radius: $rounded-rectangle-radius;
    overflow-y: hidden;
    overflow-x: auto;
  }
}
