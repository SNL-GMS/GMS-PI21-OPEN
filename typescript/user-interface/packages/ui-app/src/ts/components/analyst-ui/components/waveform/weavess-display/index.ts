import * as WeavessDisplayTypes from './types';

export { WeavessDisplay } from './weavess-display-container';
export { WeavessDisplayTypes };
