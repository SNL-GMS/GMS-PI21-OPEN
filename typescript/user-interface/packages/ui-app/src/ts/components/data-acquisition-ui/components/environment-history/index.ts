export { EnvironmentContainer as SohEnvironmentHistory } from './environment-history-container';
