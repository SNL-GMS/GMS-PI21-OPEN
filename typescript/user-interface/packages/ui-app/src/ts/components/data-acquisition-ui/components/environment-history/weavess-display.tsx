/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import type { SohTypes } from '@gms/common-model';
import type { HSL } from '@gms/ui-util';
import { DistinctColorPalette, hslToHex, UILogger } from '@gms/ui-util';
import { Weavess } from '@gms/weavess';
import { WeavessConstants, WeavessTypes, WeavessUtil } from '@gms/weavess-core';
import sortBy from 'lodash/sortBy';
import * as React from 'react';

import { dataAcquisitionUserPreferences } from '~components/data-acquisition-ui/config';

const logger = UILogger.create('GMS_SOH_WEAVESS', process.env.GMS_SOH_WEAVESS);

export const WeavessDisplay: React.FunctionComponent<{
  station: SohTypes.UiStationSoh;
  channelSohs: SohTypes.ChannelSoh[];
  sohHistoricalDurations: number[];
  startTimeMs: number;
  endTimeMs: number;
  aceiData: SohTypes.UiHistoricalAcei[];
  // eslint-disable-next-line react/function-component-definition
}> = props => {
  const { channelSohs } = props;
  const sortedChannelSohs = sortBy<SohTypes.ChannelSoh>(channelSohs, [
    channelSoh => channelSoh.channelName
  ]);
  const channelNames = React.useMemo(
    () => sortedChannelSohs.map(channel => channel.channelName).sort(),
    [sortedChannelSohs]
  );

  /** unique color palette created for the channels and station */
  const colorPalette = React.useMemo(
    () => new DistinctColorPalette(channelNames, props.station.stationName),
    [channelNames, props.station.stationName]
  );

  const stations: WeavessTypes.Station[] = [];
  sortedChannelSohs.forEach((channelSoh, i) => {
    const data = props.aceiData?.find(c => c.channelName === channelSoh.channelName);
    let channelColor: HSL;
    try {
      channelColor = colorPalette.getColor(channelSoh.channelName);
    } catch (e) {
      logger.warn(`Failed to find color for channel ${channelSoh.channelName} skipping. ${e}`);
    }

    const channelSegmentsRecord: Record<string, WeavessTypes.ChannelSegment[]> = {};
    channelSegmentsRecord.data =
      data && channelColor
        ? [
            {
              channelName: channelSoh.channelName,
              wfFilterId: WeavessTypes.UNFILTERED,
              isSelected: false,
              dataSegments: data?.issues.map<WeavessTypes.DataSegment>(issue => ({
                color: hslToHex(channelColor),
                displayType: [WeavessTypes.DisplayType.LINE],
                pointSize: 4,
                data: WeavessUtil.createStepPoints(issue)
              }))
            }
          ]
        : [];

    stations.push({
      id: String(i),
      name: `station ${channelSoh.channelName}`,
      defaultChannel: {
        height: 60,
        defaultRange: {
          min: -0.2,
          max: 1.2
        },
        yAxisTicks: [0, 1],
        id: channelSoh.channelName,
        name: `${channelSoh.channelName.replace(`${props.station.stationName}.`, '')}`,
        waveform: {
          channelSegmentId: 'data',
          channelSegmentsRecord
        }
      },
      nonDefaultChannels: undefined, // Has no child channels
      areChannelsShowing: false
    });
  });

  return (
    <div
      style={{
        height: `${dataAcquisitionUserPreferences.minChartHeightPx}px`
      }}
      className="weavess-container"
      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
      tabIndex={0}
    >
      <div className="weavess-container__wrapper">
        <Weavess
          viewableInterval={{
            startTimeSecs: props.startTimeMs / WeavessConstants.MILLISECONDS_IN_SECOND,
            endTimeSecs: props.endTimeMs / WeavessConstants.MILLISECONDS_IN_SECOND
          }}
          isControlledComponent={false}
          minimumOffset={0}
          maximumOffset={0}
          stations={stations}
          selections={{
            channels: undefined
          }}
          initialConfiguration={{
            labelWidthPx: 180,
            defaultChannel: {
              disableMeasureWindow: true,
              disableMaskModification: true
            }
          }}
          events={WeavessConstants.DEFAULT_UNDEFINED_EVENTS}
        />
      </div>
    </div>
  );
};
