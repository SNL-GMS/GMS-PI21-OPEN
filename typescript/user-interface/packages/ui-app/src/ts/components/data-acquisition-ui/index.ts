import * as DataAcquisitionComponents from './components';

export { DataAcquisitionComponents };
