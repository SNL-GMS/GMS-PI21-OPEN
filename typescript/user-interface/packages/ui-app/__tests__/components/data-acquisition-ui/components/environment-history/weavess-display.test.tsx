/* eslint-disable @typescript-eslint/no-magic-numbers */
import { SohTypes } from '@gms/common-model';
import { render } from '@testing-library/react';
import React from 'react';

import { WeavessDisplay } from '../../../../../src/ts/components/data-acquisition-ui/components/environment-history/weavess-display';
import { testStationSoh } from '../../../../__data__/data-acquisition-ui/soh-overview-data';

const MOCK_TIME = 1611153271425;
Date.now = jest.fn(() => MOCK_TIME);
Date.constructor = jest.fn(() => new Date(MOCK_TIME));

describe('Environment history panel', () => {
  it('should be defined', () => {
    expect(Date.now()).toEqual(MOCK_TIME);
    expect(WeavessDisplay).toBeDefined();
  });

  it('render weavess display', () => {
    const { container } = render(
      <WeavessDisplay
        startTimeMs={500}
        endTimeMs={1000}
        channelSohs={testStationSoh.channelSohs}
        sohHistoricalDurations={[]}
        station={testStationSoh}
        aceiData={[
          {
            channelName: testStationSoh.channelSohs[0].channelName,
            monitorType: SohTypes.AceiType.CLIPPED,
            issues: [
              [
                [1482456217000, 0],
                [1482456293000, 1],
                [1482456369000, 0],
                [1482456445000, 0]
              ],
              [
                [1482457781000, 1],
                [1482457857000, 0],
                [1482457933000, 1],
                [1482458009000, 1]
              ],
              [
                [1482459345000, 0],
                [1482459421000, 1],
                [1482459497000, 0],
                [1482459573000, 0]
              ]
            ]
          }
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
