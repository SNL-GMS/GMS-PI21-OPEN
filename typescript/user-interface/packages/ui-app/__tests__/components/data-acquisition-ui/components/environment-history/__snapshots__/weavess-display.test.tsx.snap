// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment history panel render weavess display 1`] = `
<div>
  <div
    class="weavess-container"
    tabindex="0"
  >
    <div
      class="weavess-container__wrapper"
    >
      <div
        class="weavess"
        data-cy="weavess-container"
        data-end-time="1"
        data-start-time="0.5"
        style="--weavess-background-color: #182026;"
      >
        <div
          class="horizontal-divider__bottom weavess-waveform-display"
          data-cy="weavess-waveform-display"
          style="height: 100%;"
        >
          <div
            class="weavess-zoom-handler"
            role="tab"
            tabindex="-1"
          >
            <div
              class="weavess-wp"
              style="height: 100%; position: relative; width: 100%; box-sizing: content-box; --weavess-scrollbar-track-margin: 180px;"
            >
              <canvas
                class="weavess-wp-canvas"
                style="width: calc(100% - 190px); height: calc(100% - 35px); left: 180px;"
              />
              <div
                class="weavess-wp-container"
              >
                <div
                  class="weavess-wp-container-1"
                >
                  <div
                    class="weavess-wp-container-2"
                  >
                    <div
                      class="weavess-wp-container-3"
                    >
                      <aside
                        style="width: 9007199254740991px; height: 1px; position: absolute; pointer-events: none; background: none;"
                      />
                      <div
                        class="weavess-wp-container-viewport"
                        tabindex="0"
                      >
                        <div
                          class="weavess-wp-container-viewport-content"
                        >
                          <div
                            class="station"
                            style="height: 61px;"
                          >
                            <div
                              class="channel"
                              data-cy="AAA111-channel"
                              style="height: 60px; max-height: 60px;"
                              tabindex="0"
                            >
                              <div
                                class="channel-label-container"
                                data-cy="AAA111-label"
                                style="height: 60px; width: 180px;"
                                tabindex="0"
                              >
                                <div
                                  class="label"
                                  tabindex="0"
                                >
                                  <div
                                    class="label-container"
                                  >
                                    <div
                                      class="label-container-left-parent"
                                    />
                                    <div
                                      class="label-container-content"
                                    >
                                      <div
                                        class="label-container-content-label"
                                        data-cy="channel-label-AAA111"
                                      >
                                        <div>
                                          <span
                                            class="weavess-tooltip__target label-container-content"
                                          >
                                            <span
                                              aria-haspopup="true"
                                              class="weavess-tooltip bp4-popover2-target"
                                            >
                                              <span
                                                class=""
                                                tabindex="0"
                                              >
                                                AAA111
                                              </span>
                                            </span>
                                            <div
                                              class="distance-azimuth"
                                            >
                                              <p>
                                                <span
                                                  class="label-container-content-mask-indicator"
                                                />
                                              </p>
                                            </div>
                                          </span>
                                        </div>
                                      </div>
                                      <div
                                        data-cy-contains-amplitude-markers="true"
                                        style="height: 100%;"
                                      >
                                        <div
                                          class="y-axis"
                                          data-cy="y-axis"
                                          data-max="1"
                                          data-min="-1"
                                          style="height: calc(100% - 4px); margin: 2px 0px 2px 0px;"
                                        >
                                          <svg
                                            height="100%"
                                            width="50"
                                          >
                                            <g
                                              class="y-axis__ordinate"
                                              height="100%"
                                              transform="translate(49,0)"
                                            />
                                          </svg>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="channel-content-container"
                                style="height: 60px; width: calc(100% - 180px); left: 180px;"
                              >
                                <div
                                  class="contentrenderer"
                                  tabindex="0"
                                >
                                  <div
                                    class="contentrenderer-content contentrenderer-content--sticky"
                                    style="width: calc(100% - 180);"
                                  />
                                  <div
                                    class="contentrenderer-content"
                                    data-cy-station-waveform="AAA111"
                                    style="left: 180px; width: calc(100% - 180px);"
                                  >
                                    <div
                                      class="contentrenderer-content-drag-indicator"
                                    />
                                    <div
                                      class="measure-window-selection measure-window-selection__container"
                                      style="display: none; transform: translateX(-50%);"
                                    >
                                      <div
                                        class="measure-window-selection__overlay"
                                        data-cy="measure-window-selection"
                                        draggable="true"
                                        style="display: none; left: 50%; width: 0%;"
                                      />
                                    </div>
                                    <div
                                      class="contentrenderer-content-markers"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            class="weavess-wp-container-viewport-content-markers"
                            style="width: calc(100% - 190px); left: 180px;"
                          />
                        </div>
                        <object
                          class="resize-sensor"
                          data="about:blank"
                          style="position: absolute; top: 0; left: 0; height: 100%; width: 100%; pointer-events: none; z-index: -1; opacity: 0;"
                          tabindex="-1"
                          type="text/html"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="weavess-wp-container-overlay"
                    style="width: calc(100% - 190px);"
                  >
                    <div
                      class="weavess-wp-container-overlay-cross-hair"
                    />
                    <div
                      class="weavess-wp-container-overlay-selection-area"
                    />
                  </div>
                </div>
                <div
                  class="x-axis"
                  data-cy="x-axis"
                  data-end-time="1"
                  data-start-time="0.5"
                  style="height: 35px;"
                >
                  <div
                    style="width: 100%;"
                  >
                    <svg
                      height="35"
                      style="fill: #F6F7F9;"
                      width="100%"
                    >
                      <g
                        class="x-axis-axis"
                        fill="none"
                        font-family="sans-serif"
                        font-size="10"
                        text-anchor="middle"
                      >
                        <path
                          class="domain"
                          d="M180.5,7V0.5H-9.5V7"
                          stroke="currentColor"
                        />
                      </g>
                    </svg>
                  </div>
                  <div
                    style="text-align: center;"
                  />
                </div>
              </div>
              <div
                class="weavess-wp-time-range"
                style="left: 180px;"
              >
                <span>
                  1970-01-01 00:00:00.500 + 0.5 seconds
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Toastify"
          id="Weavess-Toast-Container"
        />
      </div>
    </div>
  </div>
</div>
`;
